control-plane/connect-inject/peering_acceptor_controller.go

```diff
@@ -388,6 +406,73 @@ func (r *PeeringAcceptorController) requestsForPeeringTokens(object client.Objec
 	return []ctrl.Request{}
 }
 
+func (r *PeeringAcceptorController) getExposeServersServiceAddresses() ([]string, error) {
+	r.Log.Info("getting external address from expose-servers service", "name", r.ExposeServersServiceName)
+	var serverExternalAddresses []string
+
+	serverService := &corev1.Service{}
+	key := types.NamespacedName{
+		Name:      r.ExposeServersServiceName,
+		Namespace: r.ReleaseNamespace,
+	}
+	err := r.Client.Get(r.Context, key, serverService)
+	if err != nil {
+		return []string{}, err
+	}
+	switch serverService.Spec.Type {
+	case corev1.ServiceTypeNodePort:
+		nodes := corev1.NodeList{}
+		err := r.Client.List(r.Context, &nodes)
+		if err != nil {
+			return []string{}, err
+		}
+		if len(nodes.Items) == 0 {
+			return []string{}, fmt.Errorf("no nodes were found for scraping server addresses from expose-servers service")
+		}
+		var grpcNodePort int32
+		for _, port := range serverService.Spec.Ports {
+			if port.Name == "grpc" {
+				grpcNodePort = port.NodePort
+			}
+		}
+		if grpcNodePort == 0 {
+			return []string{}, fmt.Errorf("no grpc port was found for expose-servers service")
+		}
+		for _, node := range nodes.Items {
+			addrs := node.Status.Addresses
+			for _, addr := range addrs {
+				if addr.Type == corev1.NodeInternalIP {
+					serverExternalAddresses = append(serverExternalAddresses, fmt.Sprintf("%s:%d", addr.Address, grpcNodePort))
+				}
+			}
+		}
+		if len(serverExternalAddresses) == 0 {
+			return []string{}, fmt.Errorf("no server addresses were scraped from expose-servers service")
+		}
+		return serverExternalAddresses, nil
+	case corev1.ServiceTypeLoadBalancer:
+		lbAddrs := serverService.Status.LoadBalancer.Ingress
+		if len(lbAddrs) < 1 {
+			return []string{}, fmt.Errorf("unable to find load balancer address for %s service, retrying", r.ExposeServersServiceName)
+		}
+		for _, lbAddr := range lbAddrs {
+			// When the service is of type load balancer, the grpc port is hardcoded to 8503.
+			if lbAddr.IP != "" {
+				serverExternalAddresses = append(serverExternalAddresses, fmt.Sprintf("%s:%s", lbAddr.IP, "8503"))
+			}
+			if lbAddr.Hostname != "" {
+				serverExternalAddresses = append(serverExternalAddresses, fmt.Sprintf("%s:%s", lbAddr.Hostname, "8503"))
+			}
```

Comment:
we should get the port from the k8s service instead of hard-coding it. It's not blocking since it matches how we define the service for now, but I think it'll be more future-proof if we get it from there.

Reply:
going to do this one later!